spec/controllers/api/drafts_controller_spec.rb

```diff
@@ -79,6 +79,60 @@
     assert_equal(parsed_body['error'], 'unauthorized')
   end
 
+  it 'can not update a draft if the user does not belong to the provider list.' do
+    # The draft is created by a MMT_2 user
+    # The testeruser2 updating the document does not have MMT_2 in their available provider list, only 'LARC'
+    allow_any_instance_of(Cmr::UrsClient).to receive(:validate_mmt_token).and_return(Faraday::Response.new(status: 200, body: '{"uid":"testuser2"}', response_headers: { 'Content-Type': 'application/json; charset=utf-8' }))
+    jsonContent = { "json": {
+      "Name": "a name",
+      "LongName": "a long name",
+      "Version": "10.0"
+    }}.to_json
+    put :update, body: jsonContent, params: { id: @tool_draft.id, draft_type: "ToolDraft" }
+    assert_equal(response.status, 401)
+    parsed_body = JSON.parse(response.body)
+    assert_equal(parsed_body['error'], 'unauthorized')
+  end
+
+  it 'create a draft record.' do
+    # The draft is created by testuser and Provider is MMT_2
+    allow_any_instance_of(Cmr::UrsClient).to receive(:validate_mmt_token).and_return(Faraday::Response.new(status: 200, body: '{"uid":"testuser"}', response_headers: { 'Content-Type': 'application/json; charset=utf-8' }))
+    jsonContent = { "json": {
+      "Name": "a new draft",
+      "LongName": "long name",
+      "Version": "10.0"
+    }}.to_json
+    request.headers.merge!({ 'User' => 'testuser' })
+    request.headers.merge!({ 'Provider' => 'MMT_2' })
+    post :create, body: jsonContent, params: { draft_type: "ToolDraft" }
+    assert_equal(response.status, 200)
+    parsed_body = JSON.parse(response.body)
+    assert_equal(parsed_body['draft']['Name'], 'a new draft')
+  end
+
+  it 'retrieve a draft record' do
+    # The draft is created by testuser and Provider is MMT_2
+    # The testeruser retrieving the draft
+    allow_any_instance_of(Cmr::UrsClient).to receive(:validate_mmt_token).and_return(Faraday::Response.new(status: 200, body: '{"uid":"testuser"}', response_headers: { 'Content-Type': 'application/json; charset=utf-8' }))
+    get :show, params: { id: @tool_draft.id, draft_type: "ToolDraft" }
+    parsed_body = JSON.parse(response.body)
+    unless parsed_body.is_a?(Hash)
+      parsed_body = JSON.parse(parsed_body)
+    end
+    response_draft = parsed_body['draft']
+    draft_json = @tool_draft.draft
+    assert_equal(response_draft['a new draft'], draft_json['a new draft'])
+  end
+
+  it 'retrieve a draft with error if a user does not belong to a provider list.' do
```

Comment:
The test 'it responds unauthorized if the user does not belong to the provider list.' (line 52-59) can be removed as this is the same test, but I like the comments better in this one.